src/workerd/io/trace-stream.c++

```diff
@@ -946,7 +949,9 @@ class TailStreamTarget final: public rpc::TailStreamTarget::Server {
       if (doFulfill) {
         p = p.then(js, [&](jsg::Lock& js) { doneFulfiller->fulfill(); },
             [&](jsg::Lock& js, jsg::Value&& value) {
-          doneFulfiller->reject(KJ_EXCEPTION(DISCONNECTED, "Streaming tail session canceled"));
+          kj::Exception exception = KJ_EXCEPTION(FAILED, "Streaming tail session exception");
```

Comment:
Why not serialize value and attach it to the exception?
tbf, the JS exception really should just be the exception here, no?